### PR TITLE
Split process output into StdOut and StdErr, don't mix these

### LINKED ISSUE #7: Sprawdzanie wersji Hełma używa wyjścia diagnostycznego
Używam CI, gdzie nie kontroluję dokładnie jakie uprawnienia ma plik z kube-configiem i każde odpalenie Helma wypisuje ostrzeżenie "WARNING: Kubernetes configuration file is world-readable."
To powoduje problem przy sprawdzaniu wersji Hełma: https://github.com/kiemlicz/shelm/blob/master/src/main/scala/io/github/kiemlicz/shelm/HelmPlugin.scala#L43
Używana metoda `startProcess` zwraca zarówno stdin i stderr, stąd wersja zamiast "3.5.4" jest "3.5.4 WARNING(...)" lub "WARNING(...) 3.5.4"

Wynik: wysypanie się
```
java.lang.IllegalArgumentException: Invalid version number: WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /builds/internal/coiote/coiote-dm.tmp/CDM_K8S_KUBECONFIG
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /builds/internal/coiote/coiote-dm.tmp/CDM_K8S_KUBECONFIG
v3.5.4
	at sbt.librarymanagement.VersionNumber$.apply(VersionNumber.scala:52)
	at io.github.kiemlicz.shelm.HelmPlugin$autoImport$.$anonfun$baseHelmSettings$4(HelmPlugin.scala:44)
	at sbt.internal.util.Init$Value.$anonfun$apply$2(Settings.scala:922)
	at sbt.internal.util.EvaluateSettings.$anonfun$constant$1(INode.scala:211)
	at sbt.internal.util.EvaluateSettings$MixedNode.evaluate0(INode.scala:228)
	at sbt.internal.util.EvaluateSettings$INode.evaluate(INode.scala:170)
	at sbt.internal.util.EvaluateSettings.$anonfun$submitEvaluate$1(INode.scala:87)
	at sbt.internal.util.EvaluateSettings.sbt$internal$util$EvaluateSettings$$run0(INode.scala:99)
	at sbt.internal.util.EvaluateSettings$$anon$3.run(INode.scala:94)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
[error] java.lang.IllegalArgumentException: Invalid version
```

### PR DESCRIPTION
**Pull request by WojciechKarpiel**
Fixes https://github.com/kiemlicz/shelm/issues/7

### CODE REVIEW

File: src/main/scala/io/github/kiemlicz/shelm/BufferingProcessLogger.scala [hunk 1]
```diff
@@ -5,24 +5,32 @@ import io.github.kiemlicz.shelm.exception.HelmCommandException
 import scala.sys.process.ProcessLogger
 
 class BufferingProcessLogger extends ProcessLogger {
-  val buf: StringBuffer = new StringBuffer()
+  val stdOutBuf: StringBuffer = new StringBuffer()
+  val stdErrBuf: StringBuffer = new StringBuffer()
```
Comment (kiemlicz) [inline src/main/scala/io/github/kiemlicz/shelm/BufferingProcessLogger.scala]: There is only one thing that makes me worried:   
having split std out with std err, imagine the helm process outputting:
```
some stdout output lines
some stderr output lines
some stdout output  lines
```

You lose ordering and yield

out:  
```
some stdout output lines
some stdout output  lines
```

err:  
```
some stderr output lines
```
Which reduces readability

Propositions:
1. have two fields: 
```
  val stdOutBuf: StringBuffer = new StringBuffer() // out only
  val buf: StringBuffer = new StringBuffer() // err+out
```
During any toString-a-like methods you use err+out as previously
2. Prove that we shouldn't care about this
3. ??? ;)

**Review by kiemlicz — CHANGES_REQUESTED** [review]
Many thanks for this PR!

File: src/main/scala/io/github/kiemlicz/shelm/BufferingProcessLogger.scala (hunk 1 above)
Comment (WojciechKarpiel) [reply src/main/scala/io/github/kiemlicz/shelm/BufferingProcessLogger.scala]: >  2. Prove that we shouldn't care about this

TL;DR I think that it's not possible to preserve ordering, and that splitting the two outputs is the correct way to go

Ordering of mixed output can't be relied upon, it depends on many factors (buffering, pipe magic). You can only rely on ordering within stdout and within stderr.
For example, stdout is usually buffered, but stderr is not, this means, that if two things happen one after another you're likely to end up with log ordering like this:
```
Stderr warning about thing A
Stderr warning about thing B
Stdout output about thing A
Stdout output about thing B
```

In case of https://github.com/kiemlicz/shelm/issues/7, I believe that such output is possible:
```
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /builds/internal/coiote/coiote-dm.tmp/CDM_K8S_KUBECONFIG
v3.5.4
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /builds/internal/coiote/coiote-dm.tmp/CDM_K8S_KUBECONFIG
```
IMO it would be hard to figure out what happens here

Comment (kiemlicz) [reply src/main/scala/io/github/kiemlicz/shelm/BufferingProcessLogger.scala]: The idea was:
- we control the commands that are executed on the machine running this plugin, so I doubt it will get `very | pipey`  
- naturally the underlying process can mix, interleave, do actually anything with its output and err
- that's the reason why for all side effecting (println, exceptions) actions we want to take the input 'as is' without any modifications

Just let the user see unmodified (by shelm) output

**Review by kiemlicz — APPROVED** [review]
Thanks!
